### DISCUSSION TITLE
Drag & Drop

### DISCUSSION THREAD

**Discussion by LMSSonos** [tui-cs/Terminal.Gui#2946] [Q&A]
In the readme I saw following feature

- Keyboard and Mouse Input - Both keyboard and mouse input are supported, including support for drag & drop.

I was wondering about the drag & drop. I can drag & drop a file (on windows) into a textfield and get the full path. I was wondering, if get the Drag&Drop event globally so that when a file is dragged into the console app that I can automatically show the field so that the path is pasted into that. 

Is this possible?

Thanks

**Comment by tznind** [comment]
I think drag and dropping a file into a console will be handled by the console host itself e.g. Windows Terminal.  Most terminals will either do nothing or 'type' the path of the file dropped.  So I think what happened was:

1. You had a `TextField` selected
2. You dropped a file into your terminal
3. Windows Terminal typed the path
4. The path appeared in the `TextField`

So there is no drag and drop event as such.

In the readme when we refer to 'drag and drop' it means within the app itself  e.g. in the designer:

<details>
<summary>Dragging and dropping in Designer </summary>
<img src="https://user-images.githubusercontent.com/31306100/175789072-27fcacf0-e9f0-4543-818d-50058bb46a2f.gif"/>
</details>
